### DISCUSSION TITLE
We should be able to report invites as spam

### DISCUSSION THREAD

**Discussion by Yoric** [element-hq/element-meta#494] [Ideas]
### Your use case

Being spammed with invites happen. One of the (many) steps towards fixing it would be making it easier to report an invite as spam.

I suggest that we should rework

![Screenshot from 2022-06-28 14-24-28](https://user-images.githubusercontent.com/10190/176177743-08835149-c1e2-4284-b7d6-7f6b4379367c.png)

# Reworking the dialog itself

![Screenshot from 2022-06-29 09-43-08](https://user-images.githubusercontent.com/10190/176380801-8c15e9b0-412b-4fd2-9678-4fb4a60efef3.png)

Behavior:

- "Accept" remains unchanged;
- "Report..." does the following:
   1. Mark the invite as ignored (see below);
   2. Open the existing Report dialog to report the invite event;
   3. Open a dialog to offer the ability to:
       a. Ignore invite
           - Behavior: Closes the dialog.
       b. Ignore any future invite to this room
           - Behavior: Closes the dialog.
           - Behavior: Add room to list of ignored invites.
       c. Ignore any future invite from this user
           - Behavior: Closes the dialog.
           - Behavior: Add user to list of ignored invites.
       d. (If at least three users from the homeserver are already ignored *and* the homeserver is not the local homeserver) Ignore any future invite from this homeserver
           - Behavior: Closes the dialog.
           - Behavior: Add server to list of ignored invites.
- "Ignore" does the following:
   1. Put the invite in the list of ignored invites;
   2. If any of the following conditions is matched, open a dialog to offer the ability to:
       a. (If this is not the first invite to the room being ignored) Ignore any future invite to this room.
          - Behavior: See above.
       b. (If this is not the first invite from this user being ignored) Ignore any future invite from this user.
          - Behavior: See above.
       c. (If at least three users from the homeserver are already ignored *and* the homeserver is not the local homeserver) Ignore any future invite from this homeserver.
          - Behavior: See above.

# Giving the ability to bulk ignore

TBD

**edit** Reworked after feedback from users.

**Comment by Yoric** [comment]
Note: We're currently starting to implement a prototype of the feature without UI.

[ ] [MSC for ignoring invites](https://github.com/matrix-org/matrix-spec-proposals/pull/3840)
[ ] [Base support for ignoring invites](https://github.com/matrix-org/matrix-js-sdk/pull/2483)
[ ] Commands for ignoring invites.
[ ] Hiding ignored invites.

**Comment by turt2live** [comment]
https://github.com/matrix-org/matrix-spec-proposals/pull/4151 makes this possible. I'm closing this in favour of https://github.com/element-hq/element-meta/issues/2486 for now - it has a more detailed set of links.
